libsolidity/interface/CompilerStack.cpp

```diff
@@ -526,7 +528,7 @@ bool CompilerStack::compile(State _stopAfter)
 						if (m_generateEvmBytecode)
 							compileContract(*contract, otherCompilers);
 						if (m_generateIR || m_generateEwasm)
-							generateIR(*contract);
+							generateIR(*contract, otherYulSources);
```

Comment:
Can you avoid passing by non-const reference?

Reply:
No, in this case the non-constness is the whole reason for the parameter. The functions receiving it modify it and the caller uses the result.

Reply:
Isn't the information already available in `m_contracts`? Also I think that `IRGenerator::generate` only picks the sources from `_otherYulSources` that it really needs, so passing all of the sources from `m_contracts` to ``IRGenerator::generate` should work, shouldn't it?